horace_core/spaghetti_plot.m

```diff
@@ -109,426 +113,486 @@
+    normal_vector = cross(u1crt, u2crt);
+
+    % Check if rlp's are collinear.
+    for j = 3:nseg
+        if abs(sum(normal_vector)) >= min(opt.qbin) / 100
+            break
+        end
         u2rlp = rlp(j,:)-rlp(j+1,:);
-        u2crt = (b*u2rlp')';
-        u2crt0 = cross(u1crt,u2crt);
-        j=j+1;
+        u2crt = sqw_proj.transform_hkl_to_img(u2rlp')';
+        normal_vector = cross(u1crt, u2crt);
     end
-    % By definition, 2 lines define a plane, so we only check the case of nseg>2
-    if nseg>2
-        for i=j:nseg
-            u2rlp = rlp(i,:)-rlp(i+1,:);
-            u2crt = (b*u2rlp')';
-            norve = cross(u1crt,u2crt);
-            if abs(sum(cross(norve,u2crt0)))>1e-5
-                % Normal vectors are not parallel. Point not in plane
-                u2crt0 = [];
-                break;
-            end
+    plane_normal = normal_vector;
+
+    for i = j:nseg
+        u2rlp = rlp(i, :) - rlp(i+1, :);
+        u2crt = sqw_proj.transform_hkl_to_img(u2rlp')';
+        normal_vector = cross(u1crt, u2crt);
+
+        if abs(sum(cross(normal_vector, plane_normal))) > 1e-5 % and are coplanar.
+            plane_normal = [];
+            break
         end
+
     end
-else
-    u2crt0 = [];
 end
-if ~isempty(u2crt0)
-    fprintf('spaghetti_plot: rlp found to lie in the plane perpendicular to (%g %g %g)\n',inv(b)*u2crt0');
+
+
+if ~isempty(plane_normal)
+    fprintf('spaghetti_plot: rlp found to lie in the plane perpendicular to (%g %g %g)\n', ...
+            sqw_proj.transform_img_to_hkl(plane_normal'));
 end
 
 
 % Loop over rlp, determines the projections and make the cuts
 %------------------------------------------------------------
 xrlp = 0;
-wdisp = repmat(d2d,1,nseg);
-for i=1:nseg
-    % Choose u1 along the user desired q-direction
-    u1rlp = rlp(i+1,:)-rlp(i,:);
-    u1crt = (b*u1rlp')';
-    u1crt = u1crt./norm(u1crt);
-    % Choose u2 to be either perpendicular to the plane of all the rlp (previous determined)
-    %   or the plane defined by u1 and c* or, if u1||c*, the plane defined by u1 and a*.
-    if isempty(u2crt0)
-        u2crt = cross(u1crt,b*[0;0;1]./norm(b*[0;0;1]));
-        if sum(abs(u2crt))<min(opt.qbin)/100
-            u2crt = cross(u1crt,b*[1;0;0]./norm(b*[1;0;0]));
+wdisp = repmat(d2d, 1, nseg);
+
+bz_norm = sqw_proj.transform_hkl_to_pix([0;0;1]);
+bz_norm = bz_norm ./ norm(bz_norm);
+
+bx_norm = sqw_proj.transform_hkl_to_pix([1;0;0]);
+bx_norm = bx_norm ./ norm(bx_norm);
+
+for i = 1:nseg
+    q_dir_rlu = rlp(i+1, :) - rlp(i, :);
+
+    q_dir_abs = sqw_proj.transform_hkl_to_pix(q_dir_rlu')';
+    q_dir_abs = q_dir_abs ./ norm(q_dir_abs);
+
+    q_start_abs = sqw_proj.transform_hkl_to_pix(rlp(i, :)')';
+    q_end_abs = sqw_proj.transform_hkl_to_pix(rlp(i+1, :)')';
+
+
+    if isempty(plane_normal)
+        dqv_abs = cross(q_dir_abs, bz_norm);
+        if sum(abs(dqv_abs)) < min(opt.qbin) / 100
+            dqv_abs = cross(q_dir_abs, bx_norm);
```

Comment:
I've no idea now why I wrote this... looking at this code now it looks completely redundant. Could you check if the output is ok if these lines are replaced with `dqv_abs = plane_normal`? If it's ok then you can also remove `bz_norm` and `bx_norm`...

(I think this code might have been a vestige of the original implementation before the code to check if the `rlp` list is coplanar).